product-service/handlers/getProductById.ts

```diff
@@ -1,26 +1,35 @@
-import fetch from "node-fetch";
+import {Pool} from "pg";
 import {APIGatewayProxyHandler} from "aws-lambda";
-import products from "../products";
-import headers from "./headers";
+import {headers, internalError} from "./helpers";
+import {DB_OPTIONS} from "./db";
+
+// cached variable
+let pool
 
 export const getProductById: APIGatewayProxyHandler = async (event) => {
+    console.log(event, event.pathParameters)
+    // Check if pool already exists. Create new one if not.
+    if (!pool) {
+        pool = new Pool(DB_OPTIONS);
+    }
+
+    // Get client from pool
+    const client = await pool.connect();
+
     try {
         const productId = event.pathParameters?.productId;
 
-        const productIndex = products.findIndex(product => product.id === productId);
-
-        if (productIndex > -1) {
-            const placeholderLink = `https://jsonplaceholder.typicode.com/posts/${productIndex + 1}`;
-            const resp = await fetch(placeholderLink);
-            const data = await resp.json();
+        const {rows: products} = await client.query(
+            `select * from products p inner join stocks s on s.product_id = p.id where p.id= '${productId}'`
```

Comment:
it could lead to SQL injections I guess. Plz check "parameterized queries" section -> https://node-postgres.com/features/queries

Reply:
@SkorykV thanks! I missed this